test/ui_test_suite.rb

```diff
@@ -2732,6 +2971,454 @@ def compare_within_rounding_error(error, a1, a2)
 		puts "Test method: #{self.method_name} successful!"
 	end
 
+	# Create a user annotation
+	test 'front-end: check user annotation creation' do
+		puts "Test method: #{self.method_name}"
+
+		# log in
+		login_path = @base_url + '/users/sign_in'
+		@driver.get login_path
+		wait_until_page_loads(login_path)
+		login($test_email)
+
+		# go to the 2d scatter plot study
+		two_d_study_path = @base_url + "/study/twod-study-#{$random_seed}"
+		@driver.get two_d_study_path
+		wait_until_page_loads(two_d_study_path)
+		open_ui_tab('study-visualize')
+
+		#Create an annotation from the study page
+
+		#Click selection tab
+		select_dropdown = @driver.find_element(:id, 'create_annotations_panel')
+		select_dropdown.click
+
+		#Enable Selection
+		wait_for_render(:id, 'toggle-scatter')
+		enable_select_button = @driver.find_element(:id, 'toggle-scatter')
+		enable_select_button.click
+
+		# click box select button
+		select_button = @driver.find_element(:xpath, "//a[@data-val='select']")
+		select_button.click
+
+		# get the points in the plotly trace
+		points = @driver.find_elements(:class, 'point')
+		el1 = points[0]
+
+		# click on the first point
+		@driver.action.click_and_hold(el1).perform
+
+		# wait for web driver
+		sleep 0.5
+
+		#drage the cursor to another point and release it
+		el2 = points[-1]
+		@driver.action.move_to(el2).release.perform
+
+		#wait for plotly and webdriver
+		sleep 1.0
+
+		#send the keys for the name of the annotation
+		annotation_name = @driver.find_element(:class, 'annotation-name')
+		name = "user-#{$random_seed}"
+		annotation_name.send_keys(name)
+		# send keys to the labels of the annotation
+		annotation_labels = @driver.find_elements(:class, 'annotation-label')
+		annotation_labels.each_with_index do |annot, i|
+			annot.send_keys("group#{i}")
+		end
+
+		sleep 0.5
+
+		#create the annotation
+		submit_button = @driver.find_element(:id, 'selection-submit')
+		submit_button.click
+
```

Comment:
You should actually call wait_for_render and look for the confirmation modal here rather than the sleep statement.  While the sleep may work most of the time, it's better to have a boolean condition to govern how long to wait.